app/controllers/application_controller.rb

```diff
@@ -2,9 +2,16 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
-  before_action :set_locale
- 
-  def set_locale
-    I18n.locale = I18n.default_locale
-  end
+	before_action :set_locale
```

Comment:
I recommend you using `?locale` to change the language. Using cookies adds to much complexity and I'm not sure how it works with other gems